lisa/variable.py

```diff
@@ -54,14 +53,13 @@ def _try_convert_type(original_value: Any, new_value: Any) -> Any:
     if original_type == new_type:
         return new_value
 
-    if new_type is not str:
-        target_type = new_type
-    else:
-        target_type = original_type
-
+    target_type = new_type if new_type is not str else original_type
     try:
         if target_type is bool:
-            new_value = bool(strtobool(new_value))
+            if new_value.lower() in ("y", "yes", "t", "true", "on", "1"):
+                new_value = True
+            elif new_value.lower() in ("n", "no", "f", "false", "off", "0"):
+                new_value = False
```

Comment:
Others should raise ValueError.

Reply:
If we were re-implementing the function independently, yes, but, in this case, an exception would just result in new_value's value not being changed.

Reply:
They wouldn't know, the exception would be caught and discarded.

Reply:
ok, recalled the logic in mind.